src/libraries/System.Private.CoreLib/src/System/IO/Strategies/AsyncWindowsFileStreamStrategy.ValueTaskSource.cs

```diff
@@ -3,47 +3,39 @@
 
 using System.Buffers;
 using System.Diagnostics;
-using System.Runtime.InteropServices;
 using System.Threading;
 using System.Threading.Tasks.Sources;
-using TaskSourceCodes = System.IO.Strategies.FileStreamHelpers.TaskSourceCodes;
 
 namespace System.IO.Strategies
 {
     internal sealed partial class AsyncWindowsFileStreamStrategy : WindowsFileStreamStrategy
     {
-        /// <summary>
-        /// Type that helps reduce allocations for FileStream.ReadAsync and FileStream.WriteAsync.
-        /// </summary>
-        private sealed unsafe class ValueTaskSource : IValueTaskSource<int>, IValueTaskSource
+        /// <summary>Reusable IValueTaskSource for FileStream ValueTask-returning async operations.</summary>
+        private sealed unsafe class ValueTaskSource : ManualResetEventSlim, IValueTaskSource<int>, IValueTaskSource
```

Comment:
`ManualResetEventSlim` implements `IDisposable` but it seems that it's not being disposed anywhere?

https://github.com/dotnet/runtime/blob/f3eafd7b3be3e869cc9d1f0bc0ec5e93c40e2f17/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/AsyncWindowsFileStreamStrategy.cs#L28-L47

https://github.com/dotnet/runtime/blob/f3eafd7b3be3e869cc9d1f0bc0ec5e93c40e2f17/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/AsyncWindowsFileStreamStrategy.cs#L106-L114

Reply:
MRES.Dispose() is a nop if you don't use its WaitHandle, which I don't.

But I've removed the MRES entirely, anyway.

Reply:
thanks for the explanation!